modules/argocd/main.tf

```diff
@@ -96,6 +96,12 @@ resource "helm_release" "argocd_application" {
     })
   ]
 
+  set {
+    name  = "source.useRecurse"
+    value = each.value.use_recurse
+    type  = "boolean"
+  }
+
```

Comment:
As per https://argo-cd.readthedocs.io/en/stable/user-guide/directory/#enabling-recursive-resource-detection, directory-type applications only work for plain manifest files. So this doesn't seem to be valid.

Reply:
correct... let me double check.
my original pr did not have this part, but afair pre-commit was complaining.
i'll work on it a bit more

Reply:
updated